app/googleanalytics/pixels.py

```diff
@@ -1,35 +1,53 @@
-from urllib.parse import urlencode
-
+import os
 from flask import current_app
+from urllib.parse import quote
+
+TEST_DOMAIN = 'https://test-api.va.gov/vanotify/'
+DEV_DOMAIN = 'https://dev-api.va.gov/vanotify/'
+PERF_DOMAIN = 'https://sandbox-api.va.gov/vanotify/'
+STAGING_DOMAIN = 'https://staging-api.va.gov/vanotify/'
+PROD_DOMAIN = 'https://api.va.gov/vanotify/'
+
+NOTIFICATION_API_GA4_GET_ENDPOINT = 'ga4/open-email-tracking'
+
+GA4_PIXEL_TRACKING_NAME = 'email_open'
+GA4_PIXEL_TRACKING_SOURCE = 'vanotify'
+GA4_PIXEL_TRACKING_MEDIUM = 'email'
+
 
+def get_domain_for_environment():
+    environment = os.getenv('NOTIFY_ENVIRONMENT', 'development')
 
-def build_ga_pixel_url(
-    notification,
-    provider,
-):
-    url_params_dict = {
-        'v': '1',
-        't': 'event',
-        'tid': current_app.config['GOOGLE_ANALYTICS_TID'],
-        'cid': notification.id,
-        'aip': '1',
-        'ec': 'email',
-        'ea': 'open',
-        'el': notification.template.name,
-        'dp': f"/email/vanotify"
-        f"{'/' + notification.service.organisation.name if notification.service.organisation else ''}"
-        f"/{notification.service.name}"
-        f"/{notification.template.name}",
-        'dt': notification.subject,
-        'cn': notification.template.name,
-        'cs': provider.get_name(),
-        'cm': 'email',
-        'ci': notification.template.id,
+    ENVIRONMENT_DOMAINS = {
+        'test': TEST_DOMAIN,
+        'development': DEV_DOMAIN,
+        'perf': PERF_DOMAIN,
+        'staging': STAGING_DOMAIN,
+        'prod': PROD_DOMAIN,
     }
 
-    url_str = current_app.config['GOOGLE_ANALYTICS_URL']
-    url_params = urlencode(url_params_dict)
-    url = f'{url_str}?{url_params}'
+    return ENVIRONMENT_DOMAINS[environment]
+
+
+def build_dynamic_ga4_pixel_tracking_url(notification):
```

Comment:
I think we want annotations for notification and the return?

Reply:
Yes, when we are creating new functions we want to add annotations.

Reply:
Updated!